.github/workflows/deploy-staging.yml

```diff
@@ -0,0 +1,121 @@
+name: Test & Deploy Staging
+
+on:
+  workflow_call:
+    inputs:
+      python_version_file:
+        description: 'The location of .python-version file of the caller workflow repository'
+        default: '.python-version'
+        type: string
+      working_directory:
+        description: 'The working directory of the caller workflow repository'
+        default: '.'
+        type: string
+
+jobs:
+  python_version:
+    uses: ./.github/workflows/python-version.yml
+    with:
+      python_version_file: ${{ inputs.python_version_file }}
+
+  deploy_staging:
+    # check if the comments come from pull request and includes @deploy
+    if: contains(github.event.comment.html_url, '/pull/') && contains(github.event.comment.body, 'deploy staging')
```

Comment:
Aren't we getting rid of `deploy staging` in favour of `deploy STAGE=staging`? Or is the intention to make this change later?

Reply:
We want to slowly move away from it, but for repos that don't have all of the basic requirements (self managed IAM roles, custom domains etc.) that workflow won't work.

This will allow us to port older repos over to the mostly new system with this approach temporarily setup. 

I could make it look for `deploy STAGE=staging` but not handle custom environments if we want the consistency across repos. thoughts?

Reply:
I am +1 to using `deploy STAGE=staging` for consistency between the two workflows.